tests/model_serving/model_server/components/kserve_dsc_deployment_mode/utils.py

```diff
@@ -67,3 +68,86 @@ def patch_dsc_default_deployment_mode(
             deployment_mode=request_default_deployment_mode,
         )
         yield dsc_resource
+
+
+def wait_for_service_config_in_cm(config_map: ConfigMap, service_config: str) -> None:
+    """
+    Wait for service config to be set in inferenceservice-config configmap.
+
+    Args:
+        config_map (ConfigMap): ConfigMap object
+        service_config (str): Service configuration type ("Headless" or "Headed")
+
+    Raises:
+        TimeoutExpiredError: If service config value is not set in configmap
+    """
+    LOGGER.info(f"Waiting for {service_config} service config to be set in {config_map.name} configmap")
+    expected_value = service_config == "Headless"
+    for sample in TimeoutSampler(
+        wait_timeout=Timeout.TIMEOUT_5MIN,
+        sleep=5,
+        func=lambda: config_map.instance.data,
+    ):
+        if sample:
+            try:
+                cm_service_config = json.loads(sample.service)["serviceClusterIPNone"]
+                if cm_service_config == expected_value:
+                    LOGGER.info(f"Service config successfully set to {service_config}")
+                    break
+            except (KeyError, json.JSONDecodeError) as e:
+                LOGGER.warning(f"Error reading service config: {e}")
+                continue
+
+
+def patch_raw_default_deployment_config(
+    dsc_resource: DataScienceCluster,
+    inferenceservice_config_cm: ConfigMap,
+    request_default_deployment_config: str,
+) -> Generator[DataScienceCluster, Any, Any]:
+    """
+    Patch DataScienceCluster object with service configuration and wait for it to be set in configmap.
+
+    Args:
+        dsc_resource (DataScienceCluster): DataScienceCluster object
+        inferenceservice_config_cm (ConfigMap): ConfigMap object
+        request_default_deployment_config (str): Service configuration type ("Headless" or "Headed")
+
+    Yields:
+        DataScienceCluster: DataScienceCluster object with updated service configuration
+    """
+    if request_default_deployment_config not in ["Headless", "Headed"]:
```

Comment:
"Headless", "Headed" - save in constants and re-use

Reply:
yes did it